gm4_washing_machines/data/gm4_washing_machines/functions/wash.mcfunction

```diff
@@ -0,0 +1,122 @@
+#@s = water tank with item in first slot 
+#run from item_fill
+
+#glass
+execute if block ~ ~ ~ hopper{Items:[{Slot: 0b, id: "minecraft:white_stained_glass"}]} run function gm4_washing_machines:washing_recipes/glass
+execute if block ~ ~ ~ hopper{Items:[{Slot: 0b, id: "minecraft:orange_stained_glass"}]} run function gm4_washing_machines:washing_recipes/glass
+execute if block ~ ~ ~ hopper{Items:[{Slot: 0b, id: "minecraft:magenta_stained_glass"}]} run function gm4_washing_machines:washing_recipes/glass
+execute if block ~ ~ ~ hopper{Items:[{Slot: 0b, id: "minecraft:light_blue_stained_glass"}]} run function gm4_washing_machines:washing_recipes/glass

```

Comment:
This function has a lot of NBT checks. It would be better to move this to a loot table where you can use item tags to detect the input and drop the resulting washed item.

Reply:
I'm gonna have to look into that